net/oic/src/api/oc_core_res.c

```diff
@@ -236,6 +252,8 @@ oc_core_init_platform(const char *mfg_name, oc_core_init_platform_cb_t init_cb,
                                    core_resources[OCF_P].interfaces);
     oc_rep_set_uint(root, p, core_resources[OCF_P].properties & ~OC_PERIODIC);
 
+    /* Initialize with random values */
+    oc_core_randomize_uuid(&uuid);
     oc_gen_uuid(&uuid);
```

Comment:
I honestly have no idea why you would initialize with random values. If this is something required by OIC, why does the library not do it already (in `oc_gen_uuid` maybe?)

Reply:
I looked into https://tools.ietf.org/html/rfc4122#page-14, as the oc_gen_uuid() is implementing type 4. This indicated that except from the 8th octet and 6th octet, others should be random.
The warning from code coverage tool is due to uninitialized variable. To fix this warning and satisfy the RFC requirement, the values need to be initialized with random values. Hence the randomization.

I can certainly move this call into oc_gen_uuid(). Let me know what you think.